nvflare/app_opt/xgboost/histogram_based_v2/runners/xgb_client_runner.py

```diff
@@ -72,11 +71,15 @@ def initialize(self, fl_ctx: FLContext):
             if not isinstance(self._metrics_writer, LogWriter):
                 self.system_panic("writer should be type LogWriter", fl_ctx)
 
-    def _xgb_train(self, params: XGBoostParams, train_data: xgb.DMatrix, val_data) -> xgb.core.Booster:
+    def _xgb_train(self, num_rounds, xgb_params: dict, xgb_options: dict, train_data, val_data) -> xgb.core.Booster:
```

Comment:
for all the parameters to be passed into xgb.train method.
suggest either totally pack it as before a custom class call "XGBoostParams"
Or totally unpack it, so this method signature is similar to https://xgboost.readthedocs.io/en/stable/python/python_api.html#xgboost.train and docstring to explain each of the argument meaning.

Randomly group some into xgb_params, some standalone and some other into xgb_options is very confusing.

Reply:
The grouping is very clear. The params (Booster parameters) is passed as its own dictionary, everything else is packed into xgb_options. The params is specifically unpacked due to its importance.

Reply:
Got it, then suggest either rename to "num_boost_round" or remove this "num_round" argument to be consistent

Reply:
The num_round is a NVFlare convention. Most jobs have this parameter. So it's better to leave it alone.

Reply:
Num rounds is ok on executor, but for xgb train method I think it is better to use same name as xgb.train.

But let's do that in follow up PRs.